src/renderers/webgl/WebGLRenderLists.js

```diff
@@ -137,6 +137,23 @@ function WebGLRenderList() {
 
 	}
 
+	// remove any dangling references to make sure no memory is leaked
+	// when the count of objects in a scene is reduced
+	function cleanup() {
+
+		for ( var i = this.renderedItemCount; i < renderItems.length; i ++ ) {
+
+			var renderItem = renderItems[ i ];
```

Comment:
Why don't you use `if ( renderItem.id === null ) break;` in your code?

Reply:
Done. I did not read the PR properly.

Reply:
It seems the line is still missing...